### DISCUSSION TITLE
Question about tooltips

### DISCUSSION THREAD

**Discussion by GalaxyGamingBoy** [FabricMC/fabric-api#1668] [Mod Dev Support] [answered]
How can I add a tooltip when you press <kbd>LShift </kbd>?           
Like When you press shift more tool tip appear.           
I know in forge to get a key press you do             
`InputConstant.isKeyDown()`.          
What is it in fabric?

**Comment by Technici4n** [comment]
I use the following helper for my mod, feel free to adjust and reuse it. Obviously, calling this on the server will cause a crash.
```java
public class InputHelper {
    public static boolean isShiftPressed() {
        long handle = MinecraftClient.getInstance().getWindow().getHandle();
        return InputUtil.isKeyPressed(handle, GLFW.GLFW_KEY_LEFT_SHIFT) || InputUtil.isKeyPressed(handle, GLFW.GLFW_KEY_RIGHT_SHIFT);
    }
}
```

**Comment by modmuss50** [comment, marked as answer]
Can you not just use `Screen.hasShiftDown()` from vanilla?

**Reply by Technici4n** [reply]
Right, no clue why I added this. :smile:
